custom_components/smartir/light.py

```diff
@@ -266,30 +294,43 @@ async def async_turn_on(self, **params):
-                new_brightness = DeviceData.closest_match(target, self._brightnesses)
-                did_something = True
-                _LOGGER.debug(
-                    f"Changing brightness from {self._brightness} step {old_brightness} to {target} step {new_brightness}"
-                )
-                steps = new_brightness - old_brightness
-                if steps < 0:
-                    cmd = CMD_BRIGHTNESS_DECREASE
-                    steps = abs(steps)
-                else:
-                    cmd = CMD_BRIGHTNESS_INCREASE
-
-                if steps > 0 and cmd:
-                    # If we are heading for the highest or lowest value,
-                    # take the opportunity to resync by issuing enough
-                    # commands to go the full range.
-                    if (
-                        new_brightness == len(self._brightnesses) - 1
-                        or new_brightness == 0
-                    ):
-                        steps = len(self._colortemps)
+                new_brightness = DeviceData.closest_match(target, self._brightnesses)                
+                final_brightness = f"{self._brightnesses[new_brightness]}"
+                if (
+                    CMD_BRIGHTNESS in self._commands
+                    and isinstance(self._commands[CMD_BRIGHTNESS], dict)
+                    and final_brightness in self._commands[CMD_BRIGHTNESS]
+                ):
+                    _LOGGER.debug(
+                        f"Changing brightness from {self._brightness} to {target} using found remote command for {final_brightness}"
+                    )
                     did_something = True
+                    found_command = self._commands[CMD_BRIGHTNESS][final_brightness]
                     self._brightness = self._brightnesses[new_brightness]
-                    await self.send_command(cmd, steps)
+                    await self.send_remote_command(found_command)
+                else:
+                    _LOGGER.debug(
+                        f"Changing brightness from {self._brightness} step {old_brightness} to {target} step {new_brightness}"
+                    )
+                    steps = new_brightness - old_brightness
+                    if steps < 0:
+                        cmd = CMD_BRIGHTNESS_DECREASE
+                        steps = abs(steps)
+                    else:
+                        cmd = CMD_BRIGHTNESS_INCREASE
+
+                    if steps > 0 and cmd:
+                        # If we are heading for the highest or lowest value,
+                        # take the opportunity to resync by issuing enough
+                        # commands to go the full range.
+                        if (
+                            new_brightness == len(self._brightnesses) - 1
+                            or new_brightness == 0
+                        ):
+                            steps = len(self._brightnesses)
+                        did_something = True
```

Comment:
This line (331) needs to move back to above line 316, actually it could move back up to above line 298 where it was previously and the one on line 307 removed, since after the change both branches of the if statement are setting it unconditionally.

The one on line 290 in the original code (aligned with 307) was left over from an earlier version where I originally did exactly what you did here, and has probably misled you about the intention. It should have been removed when it was moved to line 270 in the original code.

Reply:
Moved the assignation to line 291, affecting the both branches of the if. Thanks!